stages/org.osbuild.coreos.live-artifacts.mono

```diff
@@ -0,0 +1,845 @@
+    # rootfs has been appended and confirming that initramfs and rootfs are
+    # from the same build.
+    stamppath = os.path.join(tmpinitrd_rootfs, 'etc/coreos-live-rootfs')
+    os.makedirs(os.path.dirname(stamppath), exist_ok=True)
+    with open(stamppath, 'w', encoding='utf8') as fh:
+        fh.write(version + '\n')
+
+    # Add placeholder for Ignition CPIO file.  This allows an external tool,
+    # `coreos-installer iso ignition embed`, to modify an existing ISO image
+    # to embed a user's custom Ignition config.  The tool wraps the Ignition
+    # config in a cpio.xz and write it directly into this file in the ISO
+    # image.  The cpio.xz will be read into the initramfs filesystem at
+    # runtime and the Ignition Dracut module will ensure that the config is
+    # moved where Ignition will see it. We only handle !s390x here since that's
+    # the simple case (where layered initrds are supported). The s390x case is
+    # handled lower down
+    if basearch != 's390x':
+        with open(os.path.join(tmpisoimages, 'ignition.img'), 'wb') as fdst:
+            fdst.write(bytes(IGNITION_IMG_SIZE))
+        igninfo_json = {'file': 'images/ignition.img'}
+
+    # Generate JSON file that lists OS features available to
+    # coreos-installer {iso|pxe} customize.  Put it in the initramfs for
+    # pxe customize and the ISO for iso customize.
+    features = json.dumps(get_os_features(tree=deployed_tree), indent=2, sort_keys=True) + '\n'
+    featurespath = os.path.join(tmpinitrd_base, 'etc/coreos/features.json')
+    os.makedirs(os.path.dirname(featurespath), exist_ok=True)
+    with open(featurespath, 'w', encoding='utf8') as fh:
+        fh.write(features)
+    with open(os.path.join(tmpisocoreos, 'features.json'), 'w', encoding='utf8') as fh:
+        fh.write(features)
+
+    # Add osmet files. Use a chroot here to use the same coreos-installer as is
+    # in the artifacts we are building. Use the deployed_tree as the chroot since
+    # that's the easiest thing to do.
+    for img, sector_size in [(img_metal, 512), (img_metal4k, 4096)]:
+        with loop_client.device(img, partscan=True, read_only=True, sector_size=sector_size) as loopdev:
+            img_name = os.path.basename(img)
+            print(f'Generating osmet file for {img_name} image')
+            img_checksum = checksum.hexdigest_file(loopdev, "sha256")
+            img_osmet_file = os.path.join(tmpinitrd_rootfs, f'{img_name}.osmet')
+            with Chroot(deployed_tree, bind_mounts=["/run", "/tmp"]) as chroot:
+                cmd = ['coreos-installer', 'pack', 'osmet', loopdev,
+                       '--description', os_release['PRETTY_NAME'],
+                       '--checksum', img_checksum, '--output', img_osmet_file]
+                chroot.run(cmd, check=True)
+
+    tmp_squashfs_dir = os.path.join(workdir, 'tmp-squashfs-dir')
+    os.mkdir(tmp_squashfs_dir)
+
+    # Since inputs are read-only and we want to modify we'll make a
```

Comment:
```suggestion
    # Since inputs are read-only and we want to modify it, we'll make a
 ```

Reply:
looks good. I'll include the change in the next upload.